### PR TITLE
Replace deprecated find_package for python

### PR DESCRIPTION
**Pull request by mborland**
Closes: #392

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Distributed under the Boost Software License, Version 1.0.
 # See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt
 
-find_package(PythonInterp)
+find_package(Python3)
 if (PYTHONINTERP_FOUND)
```
Comment (henryiii) [inline test/CMakeLists.txt:6]: This is not the same variable. And this requires CMake 3.15+.

Comment (henryiii) [reply test/CMakeLists.txt:6]:
```suggestion
if (Python3_FOUND)
```

This is the new variable.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
   # checks that b2 and cmake are in sync
   add_test(NAME runpy-${PROJECT_NAME}_check_build_system COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/check_build_system.py)
```
Comment (henryiii) [inline test/CMakeLists.txt:8]: You should be able to use the target here, `Python3::Interpreter`. If not, `${Python3_EXECUTABLE}` is the new variable name.

Comment (HDembinski) [reply test/CMakeLists.txt:8]: `find_package(Python3::Interpreter)` did not work, but `find_package(Python3 COMPONENTS Interpreter)` does. I will implement that in another minor PR.

Comment (henryiii) [reply test/CMakeLists.txt:8]: That’s the default so it’s fine either way, what I meant was `${Python3_EXECUTABLE}` (the variable) can be replaced by `Python3::Interpreter` (a CMake target).

Comment (henryiii) [reply test/CMakeLists.txt:8]: (After COMMAND a target is supported)

Comment (HDembinski) [reply test/CMakeLists.txt:8]: Ok, I see.

**Pull request merged**